core/src/test/scala/integration/kafka/server/KRaftClusterTest.scala

```diff
@@ -761,10 +761,7 @@ class KRaftClusterTest {
     image: ClusterImage,
     brokerId: Int
   ): Boolean = {
-    Option(image.brokers().get(brokerId)) match {
-      case None => false
-      case Some(registration) => !registration.fenced()
-    }
+    Option(image.brokers().get(brokerId)).exists(registration => !registration.fenced())
```

Comment:
Just because there is an inline annotation doesn't mean that there are no allocations. It depends on the Scala version and compilation flags. Check `scalaOptimizerMode` in the Github Readme to learn more.

Reply:
understood, I had not realized that we don't run with `inline-scala` by default. In that case, is the recommendation to never use `Option.exists` in Kafka code base or is your concern about object allocations specific to performance sensitive code path only?

Reply:
The latter. Since this PR motivated some of the changes as performance improvements, I wanted to highlight that some of them _may_ have the opposite effect.

Reply:
Fair point. I have reverted the changes in code paths which I consider performance critical.